pix2pix-ultimate/metrics.py

```diff
@@ -1,44 +1,270 @@
+    """
+    from_ = row * img_size
     to = from_ + img_size
     img_row = overall_img[from_:to]
-    return np.array(np.split(img_row, 10, axis=1))
+    return np.array(np.split(img_row, channels, axis=1))
+
+
+def prepare_images(file, batch_size=16, channels=10):
+    """ Transforms the input file to an array with the shape [batch_size, dimensions, width, height].
 
-# Output array has the shape [batch_size, dimensions, width, height]
-def prepare_images(file, batch_size = 16):
-    channels = 10  # 10 dimensions for each slice of a brain
+    :param file: string - path to the file
+    :param batch_size: integer - size/rows of a batch
+    :param channels: integer - images/channels for each row/brain
+    :return: array - prepared images with the shape [batch_size, dimensions, width, height]
+    """
     input_img = np.array(scipy.misc.imread(file, flatten=True))  # (2048, 1280)
     img_size = int(input_img.shape[1] / channels)
     slices = int(input_img.shape[0] / img_size)
     output = np.empty((slices, channels, img_size, img_size))
-    for i in range(16):
+    for i in range(batch_size):
         output[i] = get_img_row(input_img, img_size, i)
     return output
 
-def example_metric(source, target):
-    # Image of one tumor region
-    return np.abs(source.sum() - target.sum())
 
-def execute_metrics(images, metrics=[example_metric]):
+def execute_metrics(images, metrics=[example_metric], channels=3, gt_offset=0, predicted_offset=7):
+    """ Execution functions to apply multiple metric functions on images.
+
+    :param images: array - row of images to apply metric functions on
+    :param metrics: array - metric functions to apply
+    :param channels: integer - amount of channels to compare
+    :param gt_offset - offset in the image row for the ground truth images
+    :param predicted_offset - offset in the image row for the target images
+    :return: tuple(array, array) - computed metric values (slices(rows), metrics, tumor regions), name of the metric
+    """
     # For each image, metric and tumor region
-    results = np.empty((images.shape[0], len(metrics), 3))
+    results = np.empty((images.shape[0], len(metrics), channels))
+    metric_names = {}
     for i, row in enumerate(images):
-        gt = row[:3]
-        predicted = row[7:]
+        gt = row[gt_offset:channels]
+        predicted = row[predicted_offset:predicted_offset + channels]
         for j, metric in enumerate(metrics):
-            for k, gt_region, pred_region in zip(range(3), gt, predicted):
-                results[i, j, k] = metric(gt_region, pred_region)
+            for k, gt_image, predicted_image in zip(range(3), gt, predicted):
+                results[i, j, k], metric_names[j] = metric(gt_image, predicted_image)
         # TODO: Later plot the image with a table containing the metrics (for presentation)
-    return results
+    return results, metric_names
+
+
+def __surface_distances(input1, input2, voxelspacing=None, connectivity=1.0):
```

Comment:
Two leading underscores are normally intended to be used for [Name Mangling](https://docs.python.org/3/tutorial/classes.html#private-variables-and-class-local-references) in classes. Since this method does not belong to a class, you can consider removing the leading underscores as there might be some undesired site effects with them (for example, when running `from X import *`, things with one underscore or more are not imported etc).